core/templates/pages/exploration-player-page/services/hint-and-solution-modal.service.spec.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for HintAndSolutionModalService.
+ */
+
+// eslint-disable-next-line max-len
+require('pages/exploration-player-page/services/hint-and-solution-modal.service.ts');
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+// TODO(#7222): Remove usage of importAllAngularServices once upgraded to
+// Angular 8.
+import { importAllAngularServices } from 'tests/unit-test-utils';
+import { TranslatorProviderForTests } from 'tests/test.extras';
+
+describe('Hint and Solution modal service factory', () => {
+  var HintAndSolutionModalService = null;
+  var $uibModal = null;
+  importAllAngularServices();
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(
+    angular.mock.module('oppia', TranslatorProviderForTests));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+
+  beforeEach(angular.mock.inject(function($injector) {
+    HintAndSolutionModalService = $injector.get(
+      'HintAndSolutionModalService');
+    $uibModal = $injector.get('$uibModal');
+  }));
+
+  it('should open an $uibModal when clicking displayHintModal', () => {
```

Comment:
```suggestion
  it('should open a modal when clicking the display hint button', () => {
```

Reply:
Try to have spec names more dependent on the actions, rather than the code. Maybe try renaming the names in this file similarly.

Reply:
sorry I'm bad at naming things XD 
done!